File: generated-sources/java-vertx/mojang-api/src/main/java/org/openapitools/server/api/model/SkinModel.java
package org.openapitools.server.api.model;

import java.util.Objects;

/**
 * Model is an empty string for the default model (like Steve) and \&quot;slim\&quot; for the slim model (like Alex)
 **/
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Model is an empty string for the default model (like Steve) and \&quot;slim\&quot; for the slim model (like Alex)
 */
public enum SkinModel {
  
  EMPTY(""),
  
  SLIM("slim");

  private String value;

  SkinModel(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static SkinModel fromValue(String text) {
    for (SkinModel b : SkinModel.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + text + "'");
  }
}

File: generated-sources/kotlin-spring/mojang-api/src/main/kotlin/org/openapitools/model/SkinModel.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonValue
import javax.validation.Valid
import javax.validation.constraints.*

/**
* Model is an empty string for the default model (like Steve) and \"slim\" for the slim model (like Alex)
* Values: eMPTY,slim
*/
enum class SkinModel(val value: String) {

    eMPTY(""),

    slim("slim");

}

